xarray/core/types.py

```diff
@@ -295,4 +295,5 @@ def copy(
 
 GroupKey = Any
 GroupIndex = Union[int, slice, list[int]]
-T_GroupIndices = list[GroupIndex]
+T_GroupIndices = tuple[GroupIndex, ...]
+T_Bins = Union[int, Sequence, np.ndarray, pd.Index]
```

Comment:
Same here with "T_".
And also, specify the generic type of Sequence. Is it: `Sequence[int]`?